### PR TITLE
Enable panel tables on model details page

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- Enable panel tables on model details page
- Change active status dot to green
- Add a loading spinner to model details page
- Conditionally render model details tables based on data available

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Click on a model
- Click the first app
- Observe app panel open
- You can now navigate the tables to slice and dice across apps/units and machines
- Refresh the page
- Observe the loading spinner

## Details

Related: https://github.com/canonical-web-and-design/juju-squad/issues/1388

### CODE REVIEW

File: src/components/panels/UnitsPanel/UnitsPanel.js [hunk 1]
```diff
@@ -1,24 +1,63 @@
...
+  onClose,
+  entity: unitId,
+  panelRowClick,
+}) {
   const modelStatusData = useModelStatus();
   const appName = unitId?.split("/")[0];
   const unit = modelStatusData?.applications[appName]?.units[unitId];
   const app = modelStatusData?.applications[appName];
 
+  const filteredModelStatusDataByMachine = useCallback(
```
Comment (hatched) [inline src/components/panels/UnitsPanel/UnitsPanel.js:30]: Doing this all the time is very inefficient, we should move this into an optimized selector.

**Review by hatched — CHANGES_REQUESTED** [review]
This works great, navigating around them is buttery smooth 👍

File: src/components/panels/UnitsPanel/UnitsPanel.js (hunk 1 above)
Comment (hatched) [reply src/components/panels/UnitsPanel/UnitsPanel.js:30]: This is for a follow-up sometime, not for now :)

**Comment by hatched** [comment]
QA Issue:

when viewing the broken model `gomboli@external/mymodel` and then clicking on any of the machines `flannel` is listed. It's a subordinate but I don't think it should be on every machine.

Screen Shot 2020-10-13 at 2 24 58 PM

File: src/components/panels/AppsPanel/AppsPanel.js [hunk 2]
```diff
@@ -24,7 +24,12 @@ import {
 
 import "./_apps-panel.scss";
 
-export default function AppsPanel({ isActive, onClose, entity }) {
+export default function AppsPanel({
+  isActive,
+  onClose,
+  entity,
+  panelRowClick,
```
[the remark this answers is not in the document]
Comment (barrymcgee) [reply src/components/panels/AppsPanel/AppsPanel.js:31]: These panels already are, by design, tightly coupled to ModelDetails. They are the coupled implementation of the generic uncoupled `SlidePanel` component.

I investigated moving the update function into a hook but it already depends on the useQueryParams hooks which is also being used in ModelDetails to control the activeView filter. Refactoring all that at this stage is not worth the cost in my opinion.

**Comment by barrymcgee** [comment]
Flannel was slipping through as it didn't have any units to interrogate. Now resolved.

File: src/components/panels/MachinesPanel/MachinesPanel.js [hunk 3]
```diff
@@ -86,6 +90,73 @@ export default function MachinesPanel({
...
+        Object.keys(filteredModelStatusData.applications).forEach(
+          (application) => {
+            const units =
+              filteredModelStatusData.applications[application]?.units;
+
+            if (Object.entries(units).length) {
+              Object.values(units).forEach((unit) => {
+                if (
+                  unit.machine !== machineId ||
+                  !Object.entries(units).length
```
Comment (hatched) [inline src/components/panels/MachinesPanel/MachinesPanel.js:106]: Can you add a comment in the code about this as it's not clear why this is being done

**Review by hatched — APPROVED** [review]

**Comment by clagom** [comment]
Nice! Not a blocker, but I wonder if we can refresh the content of the panel instead of closing/opening it again when clicking on a unit or machine from an app panel. We refresh the content when clicking on other apps from the list instead of closing/opening the panel again, so we could apply the same. @barrymcgee @hatched

**Comment by barrymcgee** [comment]
@clagom Good call.. that change actually simplifies the code too. I've now made that change.. can you re-review?

File: src/components/panels/MachinesPanel/MachinesPanel.js (hunk 3 above)
Comment (barrymcgee) [reply src/components/panels/MachinesPanel/MachinesPanel.js:102]: Done

**Pull request merged**